packages/create-catalyst/src/utils/config.ts

```diff
@@ -0,0 +1,100 @@
+import { parse as parseTOML, stringify as stringifyTOML } from '@iarna/toml';
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
+import { dirname, join } from 'path';
+
+interface CatalystConfig {
+  auth?: {
+    storeHash?: string;
+    accessToken?: string;
+  };
+}
+
+interface TomlRecord {
+  [key: string]: string | number | boolean | TomlRecord | TomlRecord[];
+}
+
+function isRecord(value: unknown): value is Record<string, unknown> {
+  return typeof value === 'object' && value !== null;
+}
+
+function isCatalystConfig(obj: unknown): obj is CatalystConfig {
+  if (!isRecord(obj)) {
+    return false;
+  }
+
+  if ('auth' in obj) {
+    if (!isRecord(obj.auth)) {
+      return false;
+    }
+
+    const { storeHash, accessToken } = obj.auth;
+
+    return (
+      (storeHash === undefined || typeof storeHash === 'string') &&
+      (accessToken === undefined || typeof accessToken === 'string')
+    );
+  }
+
+  return true;
+}
+
+export class Config {
+  private configPath: string;
+  private config: CatalystConfig;
+
+  constructor(projectDir: string) {
+    this.configPath = join(projectDir, '.catalyst');
```

Comment:
This is being used to handle .env vars for the CLI, right? Why wouldn't we use a clearer naming scheme for the file that makes it clear it's an additional .env file. To prevent potential user error of committing it in the future.

Note: `.env-catalyst-cli.local` should be caught automatically by the main [.gitignore](https://github.com/bigcommerce/catalyst/blob/main/.gitignore) (could add another catch to make it a cleaner filename)

I am pulling from my recent experience with Sentry, where their CLI automatically creates a `.env.sentry-build-plugin`:
```
# DO NOT commit this file to your repository!
# The SENTRY_AUTH_TOKEN variable is picked up by the Sentry Build Plugin.
# It's used for authentication when uploading source maps.
# You can also set this env variable in your own `.env` files and remove this file.
SENTRY_AUTH_TOKEN=***********
```

Reply:
This filename was chosen a while ago and is guaranteed to be in `.gitignore` for all previous users. I think we'd probably name it slightly differently today (this was inspired by `.stencil`) but I think it's better to stick with this previous decision.

https://github.com/bigcommerce/catalyst/pull/575

Agree we should add some preamble to the file though, good idea

Reply:
Preamble added.